### PR TITLE
`Dropdown::ListItem::Interactive` enhancements MAIN

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR will enhance the `Dropdown::ListItem::Interactive` by adding a yielding block.

### :hammer_and_wrench: Detailed description

This PR contains the following previously approved child PRs
- [Component enhancements](https://github.com/hashicorp/design-system/pull/2346)
- [Code doc changes](https://github.com/hashicorp/design-system/pull/2349)
- [Codemod](https://github.com/hashicorp/design-system/pull/2331)
- [Design doc changes](https://github.com/hashicorp/design-system/pull/2350#event-13981195668)

### :link: External links

Jira ticket: [HDS-3620](https://hashicorp.atlassian.net/browse/HDS-3620)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3620]: https://hashicorp.atlassian.net/browse/HDS-3620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md
Comment (KristinLBradley) [inline packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md:24]: The HDS team normally uses double quotes within HTML and components to enclose argument values.

File: .changeset/nice-drinks-do.md
Comment (alex-ju) [inline .changeset/nice-drinks-do.md:6]: Don't forget to update the changeset – otherwise I think we're ready to merge this

File: .changeset/nice-drinks-do.md
Comment (MelSumner) [inline .changeset/nice-drinks-do.md:9]: Question: why deprecate `@text`? Doesn't that make this component different from the same named components _not_ in the dropdown?

File: .changeset/nice-drinks-do.md
Comment (MelSumner) [inline .changeset/nice-drinks-do.md:6]: Question: if you're including a code mod, should you include that in the changeset notes?

**Review by MelSumner — COMMENTED** [review]
I left a couple of questions but otherwise it seems fine. 👍

**Review by majedelass — APPROVED** [review]
Approving after looking over the docs (via vercel website) and visually checking the component via vercel's showcase too. There don't appear to be any issues from my end :shipit:

File: .changeset/nice-drinks-do.md
Comment (zamoore) [reply .changeset/nice-drinks-do.md:9]: Are you asking about if it differs from `Hds::Interactive`? That component doesn't accept `@text` either, so one of the motivators was to make the API the same afaik.

File: packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md
Comment (zamoore) [reply packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md:19]: Can you clarify what you mean? The codemod should work fine on those examples.

In the first example:

```
<Hds::Dropdown
  class="hds-side-nav__dropdown"
  @listPosition="bottom-left"
  data-test-help-menu
  as |dd|
>
  <dd.Interactive
    @href="https://developer.hashicorp.com/hcp/docs"
    @text={{t "components.help-menu.links.documentation"}}
  />
</Hds::Dropdown::Interactive>
```

We'll find the interactive element by locating the Dropdown parent node:

```
if (node.type === 'ElementNode' && node.tag === 'Hds::Dropdown') {
```

and then iterate over its children looking for yielded interactive components:

```
if (child.type === 'ElementNode' && child.tag === `${asPrefix}.Interactive`) {
```

File: packages/components/src/components/hds/dropdown/list-item/interactive.ts
Comment (zamoore) [reply packages/components/src/components/hds/dropdown/list-item/interactive.ts:54]: Good catch. I had a URL here that pointed to the docs that must have gotten lost in a rebase. Let me fix.

File: packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md
Comment (didoo) [reply packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md:19]: I may be wrong, but I did a quick check and this is what I see:
screenshot_4103

Comment (zamoore) [reply packages/codemods/transforms/v4/dropdown-list-item-interactive/README.md:19]: Ah! I see what you mean. HDS content for the @text value. I did not consider this. Good catch. Let me see what I can do.

**Review by LilithJames-HDS — APPROVED** [review]

**Review by didoo — APPROVED** [review]
I haven't reviewed everything, just made sure that the tests of the codemod were covering enough use cases. From what I see now the coverage is good (and I can't think if something extra not covered) so I am approving that part (and defer to the other reviewers to cover the whole PR). Good job (and particularly appreciated that you gave meaningful names to the test files 🙌)

**Comment by Dhaulagiri** [comment]
🎉  nice one @LilithJames-HDS @zamoore

**Pull request merged**
